File: C#/3. Starbuzz (Decorator)/Classes/Beverage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Starbuzz.Classes
{
    public abstract class Beverage
    {  
        public enum Size { TALL, GRANDE, VENTI };

        protected string _Description { get; set; } = "Unknown Beverage";
        protected Size _Size = Size.TALL;
        public Size Sizes
        {
            get 
            { 
                return _Size;
            }

            set
            {
                _Size = value;
            }
        }

        public virtual string GetDescription()
        {
            return _Description;
        }

        public abstract double Cost();
        
    }
}

File: Java/3 - Starbuzz/src/main/java/com/mycompany/starbuzz/Beverage.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.starbuzz;

/**
 *
 * @author ismas
 */
public abstract class Beverage {
    String description = "Unknown Beberage";
    
    public enum Size { TALL, GRANDE, VENTI };
    Size size = Size.TALL;
    
    public String getDescription(){
        return description;
    }
    
    public abstract double cost();
    
    public void setSize( Size size ){
        this.size = size;
    }
    
    public Size getSize(){
        return this.size;
    }
}
